### PR TITLE
VizBuilder tabs and json edit boxes

### PR DESCRIPTION
**Pull request by tcaiger**
### Issue #:
- 3048 Json Edit Boxes https://github.com/beyondessential/tupaia-backlog/issues/3048
- 3044 Tabs https://github.com/beyondessential/tupaia-backlog/issues/3044

### Changes:

- add custom tabs
- add json edit components using 'jsoneditor-react'
- add panel to hold tabs

---

### Screenshots:
see [here](https://github.com/beyondessential/tupaia-backlog/issues/3048)

### CODE REVIEW

File: packages/admin-panel/src/VizBuilderApp/components/Panel.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const Tabs = styled(MuiTabs)`
+  width: 100%;
+  padding-right: 1.6rem;
+
+  .MuiTabs-indicator {
+    height: 5px;
+  }
+`;
+
+const Tab = styled(MuiTab)`
```
Comment (tcaiger) [inline packages/admin-panel/src/VizBuilderApp/components/Panel.js:39]: These tabs are pretty custom so I didn't bother making them into their own component file

File: packages/admin-panel/src/VizBuilderApp/components/PlayButton.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  .MuiSvgIcon-root {
+    width: 1em;
+    height: 1em;
+  }
+`;
+
+export const PlayButton = ({ setEnabled }) => {
+  const handleClick = () => {
+    setEnabled(true);
```
Comment (tcaiger) [inline packages/admin-panel/src/VizBuilderApp/components/PlayButton.js:25]: Once the `fetchPreviewData` is setup the enabled state will be used to trigger the query

File: packages/admin-panel/src/VizBuilderApp/components/TabPanel.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  if (!isSelected) {
+    return null;
+  }
+  return <Panel>{children}</Panel>;
+});
+
+TabPanel.propTypes = {
+  children: PropTypes.node.isRequired,
+  isSelected: PropTypes.bool,
+  Panel: PropTypes.any,
```
Comment (kael89) [inline packages/admin-panel/src/VizBuilderApp/components/TabPanel.js:18]:
```suggestion
  Panel: PropTypes.node,
```

File: packages/admin-panel/src/VizBuilderApp/components/PlayButton.js (hunk 2 above)
Comment (kael89) [reply packages/admin-panel/src/VizBuilderApp/components/PlayButton.js:25]: Great!

**Review by kael89 — APPROVED** [review]
Awesome work @tcaiger !

**Pull request merged**